tests/unit_tests/simtel/test_simtel_config_writer.py

```diff
@@ -132,3 +134,41 @@ def test_convert_model_parameters_to_simtel_format(
     with open(Path(model_path) / value) as f:
         content = f.read()
         assert "Trigger 2 of 1" in content
+
+
+def test_get_simtel_metadata_with_model_parameters(simtel_config_writer):
+    model_parameters = {"test_param": {"value": 42, "meta_parameter": True}}
+
+    def mock_get_name(key, simulation_software, set_meta_parameter):
+        if simulation_software == "sim_telarray":
+            if set_meta_parameter:
+                return "test_param_meta"
+            return "test_param"
+        return None
+
+    with mock.patch(
+        "simtools.utils.names.get_simulation_software_name_from_parameter_name",
+        side_effect=mock_get_name,
+    ):
+        tel_meta = simtel_config_writer._get_simtel_metadata("telescope", model_parameters)
+        assert "metaparam telescope add test_param" in tel_meta
+        assert "metaparam telescope set test_param_meta=42" in tel_meta
+
+        site_meta = simtel_config_writer._get_simtel_metadata("site", model_parameters)
+        assert "metaparam global add test_param" in site_meta
+        assert "metaparam global set test_param_meta=42" in site_meta
+
+
+def test_get_simtel_metadata_without_model_parameters(simtel_config_writer):
+    _tel = simtel_config_writer._get_simtel_metadata("telescope", None)
+    assert len(_tel) == 8
+    assert f"camera_config_name = {simtel_config_writer._telescope_model_name}" in _tel
+    assert f"optics_config_name = {simtel_config_writer._telescope_model_name}" in _tel
+
+    _site = simtel_config_writer._get_simtel_metadata("site", None)
+    assert len(_site) == 8
```

Comment:
I suggest to remove this assert, it's not very useful and will just be annoying if in the future we add/remove a parameter.

Reply:
Removed.